osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,672 @@
+      (symbols->GetFieldOffset(
+           ntdllBase, pebTypeId, "ProcessParameters", &procParamsOffset) !=
+       S_OK)) {
+    return Status(1);
+  }
+  // Get address of ProcessParameters struct
+  unsigned long long procParamsAddr = 0;
+  if (data->ReadPointersVirtual(
+          1, pebAddr + procParamsOffset, &procParamsAddr) != S_OK) {
+    return Status(1);
+  }
+
+  // Get CurrentDirectory offset in ProcessParameters
+  unsigned long procParamsTypeId = 0;
+  unsigned long curDirOffset = 0;
+  if ((symbols->GetTypeId(ntdllBase,
+                          "_RTL_USER_PROCESS_PARAMETERS",
+                          &procParamsTypeId) != S_OK) ||
+      (symbols->GetFieldOffset(
+           ntdllBase, procParamsTypeId, "CurrentDirectory", &curDirOffset) !=
+       S_OK)) {
+    return Status(1);
+  }
+  // Log CurrentDirectory
+  unsigned long long curDirBufferAddr = 0;
+  if (data->ReadPointersVirtual(
+          1, procParamsAddr + curDirOffset + 0x8, &curDirBufferAddr) != S_OK) {
+    return Status(1);
+  }
+  wchar_t curDir[MAX_PATH + 1] = {0};
+  data->ReadUnicodeStringVirtualWide(
+      curDirBufferAddr, sizeof(curDir), curDir, MAX_PATH + 1, nullptr);
+  r["current_directory"] = wstringToString(curDir);
+
+  // Get CommandLine offset in ProcessParameters
+  unsigned long cmdLineOffset = 0;
+  if (symbols->GetFieldOffset(
+          ntdllBase, procParamsTypeId, "CommandLine", &cmdLineOffset) != S_OK) {
+    return Status(1);
+  }
+  // Log CommandLine
+  unsigned long long cmdLineBufferAddr = 0;
+  if (data->ReadPointersVirtual(1,
+                                procParamsAddr + cmdLineOffset + 0x8,
+                                &cmdLineBufferAddr) != S_OK) {
+    return Status(1);
+  }
+  wchar_t cmdLine[UNICODE_STRING_MAX_BYTES] = {0};
+  data->ReadUnicodeStringVirtualWide(cmdLineBufferAddr,
+                                     sizeof(cmdLine),
+                                     cmdLine,
+                                     UNICODE_STRING_MAX_BYTES,
+                                     nullptr);
+  r["command_line"] = wstringToString(cmdLine);
+
+  // Get Environment offset in ProcessParameters
+  unsigned long envOffset = 0;
+  if (symbols->GetFieldOffset(
+          ntdllBase, procParamsTypeId, "Environment", &envOffset) != S_OK) {
+    return Status(1);
+  }
+  // Get Environment
+  unsigned long long envBufferAddr = 0;
+  if (data->ReadPointersVirtual(
+          1, procParamsAddr + envOffset, &envBufferAddr) != S_OK) {
+    return Status(1);
+  }
+  // Loop through environment variables and log those of interest
+  // The environment variables are stored in the following format:
+  // Var1=Value1\0Var2=Value2\0Var3=Value3\0 ... VarN=ValueN\0\0
+  wchar_t env[UNICODE_STRING_MAX_BYTES] = {0};
+  unsigned long bytesRead = 0;
+  data->ReadUnicodeStringVirtualWide(
+      envBufferAddr, sizeof(env), env, UNICODE_STRING_MAX_BYTES, &bytesRead);
+  while (bytesRead > sizeof(wchar_t)) {
```

Comment:
This looks a little strange for a comparison. How about capturing the return value of `data->ReadUnicodeStringVirtualWide` above and then looping while that output is good? Maybe something like:
```
  auto ret = data->ReadUnicodeStringVirtualWide(
                      envBufferAddr, sizeof(env), env, UNICODE_STRING_MAX_BYTES, &bytesRead);
  while (ret == S_OK) {
    ...
      ret = data->ReadUnicodeStringVirtualWide(envBufferAddr,
                                           sizeof(env),
                                           env,
                                           UNICODE_STRING_MAX_BYTES,
                                           &bytesRead);
  }
```